# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //base/startup/init_lite

param_service: Add to build system

In OpenHarmony 3.0 Param Service is a part of the whole Init system. This
service is required by many other services. Therefore we have to split out
the service as an independent from Init when Init is not used.

Signed-off-by: Robert Drab <robert.drab@huawei.com>
Upstream-Status: Inappropriate

diff --git a/services/BUILD.gn b/services/BUILD.gn
index 1994c2ef..e094f7ce 100644
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -85,6 +85,25 @@ if (defined(ohos_lite)) {
 } else {
   import("//build/ohos.gni")
 
+  ohos_executable("param_service") {
+    sources = [ "param/src/param_service_main.c" ]
+    include_dirs = [
+      "//base/startup/init_lite/services/include/param",
+      "//base/startup/init_lite/services/log",
+      "//third_party/cJSON",
+    ]
+    deps = [
+      "//base/startup/init_lite/services/log:init_log",
+      "//base/startup/init_lite/services/param:paramservice",
+    ]
+    install_images = [
+      "system",
+      "updater",
+    ]
+    install_enable = true
+    part_name = "startup_l2"
+  }
+
   ohos_executable("init") {
     sources = [
       "src/device.c",
@@ -132,6 +151,7 @@ if (defined(ohos_lite)) {
 
   group("startup_init") {
     deps = [
+      ":param_service",
       ":init",
       ":init_etc",
       "//base/startup/init_lite/interfaces/innerkits/socket:libsocket",
@@ -180,7 +200,7 @@ if (defined(ohos_lite)) {
 
   ohos_prebuilt_etc("ohos.para") {
     source = "//base/startup/init_lite/services/etc/ohos.para"
-    part_name = "init"
+    part_name = "startup_l2"
   }
 
   group("init_etc") {
diff --git a/services/param/BUILD.gn b/services/param/BUILD.gn
index 07e1d278..ba420b66 100644
--- a/services/param/BUILD.gn
+++ b/services/param/BUILD.gn
@@ -83,7 +83,7 @@ ohos_executable("getparam") {
     "//third_party/cJSON:cjson_static",
   ]
   install_enable = true
-  part_name = "init"
+  part_name = "startup_l2"
 }
 
 ohos_executable("setparam") {
@@ -100,5 +100,5 @@ ohos_executable("setparam") {
     "//third_party/cJSON:cjson_static",
   ]
   install_enable = true
-  part_name = "init"
+  part_name = "startup_l2"
 }
diff --git a/services/param/service/param_service.c b/services/param/service/param_service.c
index cd5d93be..03a10ae0 100644
--- a/services/param/service/param_service.c
+++ b/services/param/service/param_service.c
@@ -131,7 +131,11 @@ static int ProcessParamSet(const RequestMsg *msg)
     ret = WritePersistParam(info[0].value, info[1].value);
     PARAM_CHECK(ret == 0, return ret, "Failed to set param");
     // notify event to process trigger
+// TODO: add some flag or something instead of if 0
+// The following call is a part of OpenHarmony init system
+#if 0
     PostTrigger(EVENT_PROPERTY, msg->content, msg->contentSize);
+#endif
     return 0;
 }
 
@@ -267,7 +271,11 @@ int SystemWriteParam(const char *name, const char *value)
     PARAM_CHECK(ret == 0, return ret, "Failed to set persist param %s", name);
 
     // notify event to process trigger
+// TODO: add some flag or something instead of if 0
+// The following call is a part of OpenHarmony init system
+#if 0
     PostParamTrigger(name, value);
+#endif
     return ret;
 }
 
